src/lib/components/Widget.js

```diff
@@ -112,11 +113,38 @@ export function Widget(props) {
     }
     if (srcOrCode?.src) {
       const src = srcOrCode.src;
+      let version = propsVersion; // may be undefined which means 'latest'
+      if (version) {
+        // console.log(`attempted to lock ${src} to ${version}`);
+        // check that the version provided matches a found version
+        const componentVersionRes = cache.cachedViewCall(
+          near,
+          near.config.contractName,
+          "keys",
+          {
+            keys: [src],
+            options: {
+              return_type: "History",
+            },
+          },
+          "final"
+        );
+
+        // drill down in response to the version array
+        const srcTokens = src.split("/");
+        const componentVersions = srcTokens.reduce((acc, curr) => {
+          return acc[curr];
+        }, componentVersionRes);
+        // console.log("component versions", componentVersions);
+        if (!componentVersions.includes(version)) {
```

Comment:
I don't think this is quite necessary to check history, and this is also an expensive call. Probably the API can return a blockHeight at which the data was stored. So we can compare it this way if it's required.

Reply:
sounds good, I was leaning towards throwing an error if a matching version wasn't found to catch user error of specifying an incorrect block, but it doesn't need to be done here. If we care about that, we could do it during editing